cli/admin/src/main/java/io/pravega/cli/admin/bookkeeper/BookKeeperDisableCommand.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.cli.admin.bookkeeper;
+
+import io.pravega.cli.admin.CommandArgs;
+import io.pravega.common.util.RetriesExhaustedException;
+import io.pravega.common.util.Retry;
+import io.pravega.segmentstore.storage.DataLogWriterNotPrimaryException;
+import java.util.concurrent.atomic.AtomicInteger;
+import lombok.Cleanup;
+import lombok.val;
+
+/**
+ * Disables a BookKeeperLog.
+ */
+public class BookKeeperDisableCommand extends BookKeeperCommand {
+    private static final int MAX_RETRIES = 10;
+    private static final Retry.RetryAndThrowBase<? extends Exception> DISABLE_RETRY = Retry
+            .withExpBackoff(100, 2, MAX_RETRIES, 1000)
+            .retryWhen(ex -> ex instanceof DataLogWriterNotPrimaryException);
+
+    /**
+     * Creates a new instance of the BookKeeperDisableCommand.
+     *
+     * @param args The arguments for the command.
+     */
+    public BookKeeperDisableCommand(CommandArgs args) {
+        super(args);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(1);
+        int logId = getIntArg(0);
+
+        @Cleanup
+        val context = createContext();
+        @Cleanup
+        val log = context.logFactory.createDebugLogWrapper(logId);
+
+        // Display a summary of the BookKeeperLog.
+        val m = log.fetchMetadata();
+        outputLogSummary(logId, m);
+        if (m == null) {
+            // Nothing else to do.
+            return;
+        } else if (!m.isEnabled()) {
+            output("BookKeeperLog '%s' is already disabled.", logId);
+            return;
+        }
+
+        output("BookKeeperLog '%s' is about to be DISABLED.", logId);
+        output("\tIts SegmentContainer will shut down and it will not be able to restart until re-enabled.");
+        output("\tNo request on this SegmentContainer can be processed until that time (OUTAGE ALERT).");
```

Comment:
I don't remember when I wrote this originally. Wasn't there supposed to be some code here to wait for user input to proceed? Giving this warning and immediately proceeding with disabling it may render the warning useless.

Reply:
You are right. Initially, I removed this additional security question to make testing simpler, but I just added it again and mocked the content of `System.in` in the tests.